cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-transform-archetype/src/main/resources/archetype-resources/src/main/java/MyTransformPlugin.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package $package;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * ETL Transform Plugin Shell - This provides a good starting point for building your own Transform Plugin
+ * For full documentation, check out: http://docs.cask.co/cdap/current/en/hydrator-manual/developing-plugins/index.html
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name("MyTransform") // <- NOTE: The name of the plugin should match the name of the docs and widget json files.
+@Description("This is my transform")
+public class MyTransformPlugin extends Transform<StructuredRecord, StructuredRecord> {
+  // If you want to log things, you will need this line
+  private static final Logger LOG = LoggerFactory.getLogger(MyTransformPlugin.class);
+
+  // Usually, you will need a private variable to store the config that was passed to your class
+  private final MyConfig config;
+
+  // This is only required for testing.
+  public MyTransformPlugin(MyConfig config) {
+    this.config = config;
+  }
+
+  /**
+   * This function is called when the pipeline is published. You should use this for validating the config and setting
+   * additional parameters in pipelineConfigurer.getStageConfigurer(). Those parameters will be stored and will be made
+   * available to your plugin during runtime via the TransformContext. Any errors thrown here will stop the pipeline
+   * from being published.
+   * @param pipelineConfigurer Configures an ETL Pipeline. Allows adding datasets and streams and storing parameters
+   * @throws IllegalArgumentException If the config is invalid.
```

Comment:
`@param` and `@throws` do not end with a period unless they are more than a single sentence. Nor do they start with a capital letter. In general, they should be short enough to not require an additional sentence. If so, consider adding the text to the main body above. Revised:
```
@throws IllegalArgumentException if the config is invalid